### PR TITLE
feat: add an additional Caddyfile config template to the existing template

### PR DESCRIPTION
**Pull request by couetilc**
This could be a potential fix for #2 , not sure if you all would prefer to add to the existing template, or replace it altogether. There could be separate env var for each option. What do you think of these changes? (explanation in README changes)

### CODE REVIEW

**Comment by couetilc** [comment]
I could also add a `CADDY_FILE` envvar that will replace the default Caddyfile.tmpl in a similar manner to how `CADDY_SNIPPET` is appended to the default Caddyfile. So there would be two options for custom templates: `CADDY_FILE` which will replace the default template, and `CADDY_SNIPPET` which will append to the existing template. Unfortunately, I do not think it's easy to support adding multiple snippets.

File: README.md [hunk 1]
```diff
@@ -34,6 +34,23 @@ Options to configure:
 
 Password should be a string `base64` encoded from `bcrypt` hash. You can use https://bcrypt-generator.com/ with default config and https://www.base64encode.org/.
 
+To include a custom template:
+- set the environment variable CADDY_SNIPPET to a file containining your custom
```
Comment (sobolevn) [inline README.md:38]:
```
`CADDY_SNIPPET`
```

File: README.md [hunk 2]
```diff
@@ -34,6 +34,23 @@ Options to configure:
...
+  Caddyfile template.
+- pass your file into the container at the path specified by envvar CADDY_SNIPPET
```
Comment (sobolevn) [inline README.md:40]:
```
`CADDY_SNIPPET`
```

File: README.md [hunk 3]
```diff
@@ -34,6 +34,23 @@ Options to configure:
...
+- it will be appended to the existing template when the container loads.
+
+Example:
+
+```yaml
+services:
+  caddy-gen:
+    volumes:
+      - ./caddy:/tmp/caddy
```
Comment (sobolevn) [inline README.md:49]: Maybe we can use the full path to `/tmp/caddy/snippet` as a volume to match the example?

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -4,6 +4,7 @@ set -o errexit
 set -o nounset
 
 # Create initial configuration:
+cat $CADDY_SNIPPET >> /code/docker-gen/templates/Caddyfile.tmpl
```
Comment (sobolevn) [inline docker-entrypoint.sh:7]: `"$CADDY_SNIPPET"`

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot! 

Two questions:
1. Is it possible to test this easily?
2. Can you please rebase you PR on `master`?

File: README.md (hunk 3 above)
Comment (couetilc) [reply README.md:52]: I've found docker expects volumes to be directories, plus I've updated the example to mount two files. The key here is CADDY_TEMPLATE and CADDY_SNIPPET have to reference a file in the container's filesystem, however the user wants to get those files on there.

**Comment by couetilc** [comment]
I'll think of a way to test, I'll updated the `.travis.yml` file

**Comment by couetilc** [comment]
@sobolevn I rebased with master and added tests to the CI, any feedback is welcome.

File: docker-entrypoint.sh [hunk 5]
```diff
@@ -1,9 +1,19 @@
...
+# caddy_template is by default set to the backup to avoid adding the snippet
+# multiple times to the same template, for example when a container restarts
+export CADDY_TEMPLATE="${CADDY_TEMPLATE:-./docker-gen/templates/Caddyfile.bkp}"
+export CADDY_SNIPPET="${CADDY_SNIPPET}"
+
 set -o errexit
 set -o nounset
 
 # Create initial configuration:
+if [[ -z "$CADDY_SNIPPET" ]]; then
```
Comment (sobolevn) [inline docker-entrypoint.sh:12]: ``[[ -z`` is a bashism, but we are using `sh` 
We can try to use http://manpages.ubuntu.com/manpages/bionic/man1/checkbashisms.1.html

**Review by sobolevn — APPROVED** [review]
Now, please you review my `sh` style fixes.
If that's ok, I would merge it right away! 👍

**Comment by sobolevn** [comment]
Looks like I broke something 😢

**Comment by couetilc** [comment]
Looks good I think we can merge, thank you for the feedback @sobolevn . I will update docs today for #52 demonstrating how to set up a file_server with the new options, then rebase, and add a feature to include snippets using container labels. We should be good to release after that.

**Review by sobolevn — APPROVED** [review]
Thanks! Great work!

**Pull request merged**